### PR TITLE
Enable AV1 encoder in DG2

### PR DESCRIPTION
**Pull request by harvan**
Tracked-On: OAM-123201

### CODE REVIEW

File: groups/codec2/true/auto_hal.in [hunk 1]
```diff
@@ -1,8 +1,67 @@
...
+    "0x56b0"
+    "0x56b1"
+    "0x56b2"
+    "0x56b3"
+    "0x56ba"
+    "0x56bb"
+    "0x56bc"
+    "0x56bd"
+    "0x56be"
+    "0x56bf"
```
Comment (feijiang1) [inline groups/codec2/true/auto_hal.in:39]: can one line include multiple gpu id, then remove lines?

File: groups/codec2/true/auto_hal.in [hunk 2]
```diff
@@ -1,8 +1,67 @@
...
+)
+
+has_intel_dg2() {
+	local ret=0
+	for f in /sys/class/drm/card*; do
```
Comment (feijiang1) [inline groups/codec2/true/auto_hal.in:44]: we only need check card0 for only dgpu exist case. If both dgpu and igpu exist, still use igpu for codec.

File: groups/codec2/true/auto_hal.in (hunk 1 above)
Comment (harvan) [reply groups/codec2/true/auto_hal.in]: Modified

File: groups/codec2/true/auto_hal.in (hunk 2 above)
Comment (harvan) [reply groups/codec2/true/auto_hal.in]: Modified

**Review by feijiang1 — APPROVED** [review]
LGTM

**Review by feijiang1 — APPROVED** [review]
LGTM

**Review by bing8deng — APPROVED** [review]

**Pull request merged**